### ISSUE TITLE
Enhance file permission unit test

### ISSUE THREAD

**Issue by squakez** [apache/camel-k#2563]
> Originally i opened that PR. But in the end they were arguing on thing i didn't see on our repo..

Yeah, I think the problem is caused by how the CI is building the application. The best approach is indeed to work on improving the test in order to make it OS agnostic by creating a file and setting some privilege to deny access. I created it some time ago and I did not consider the impact on other systems that can build the application.

I am opening a follow up issue to work on such improvement, in the while, we may even remove/comment the test in order to fix the problem described there.

_Originally posted by @squakez in https://github.com/apache/camel-k/issues/2561#issuecomment-896015636_

**Comment by lzap** [comment]
Looks like the test was already fixed, the Go build tag can be released now: https://github.com/apache/camel-k/pull/5380

**Comment by squakez** [comment]
> Looks like the test was already fixed, the Go build tag can be released now: #5380

Thanks for having a look. However I think the main problem of this test was the failure on Windows OS where it is still disabled.

**Comment by lzap** [comment]
> Thanks for having a look. However I think the main problem of this test was the failure on Windows OS where it is still disabled.

Hello, not sure if I follow. Isn’t this tested now in the test code itself?

```
	if isWindows() {
		t.Skip("Test not reliably producing a result on a windows OS")
	}
```

**Comment by squakez** [comment]
> > Thanks for having a look. However I think the main problem of this test was the failure on Windows OS where it is still disabled.
> 
> Hello, not sure if I follow. Isn’t this tested now in the test code itself?
> 
> ```
> 	if isWindows() {
> 		t.Skip("Test not reliably producing a result on a windows OS")
> 	}
> ```

No, the test is skipped when the build is running in a Windows machine. The idea of this issue is to rework the test in order to allow it running regardless the operating system.

**Comment by lzap** [comment]
Oh I see, well, I do not have a Windows system here. Let’s keep it opened then, thanks.
